bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerHttpBridgeHandler.java

```diff
@@ -0,0 +1,82 @@
+package org.openhab.binding.opensprinkler.internal.handler;
+
+import static org.openhab.binding.opensprinkler.internal.OpenSprinklerBindingConstants.DEFAULT_REFRESH_RATE;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApi;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApiFactory;
+import org.openhab.binding.opensprinkler.internal.api.exception.CommunicationApiException;
+import org.openhab.binding.opensprinkler.internal.api.exception.GeneralApiException;
+import org.openhab.binding.opensprinkler.internal.config.OpenSprinklerHttpInterfaceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt - Refactoring
+ */
+@NonNullByDefault
+public class OpenSprinklerHttpBridgeHandler extends OpenSprinklerBaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(OpenSprinklerHttpBridgeHandler.class);
+
+    @Nullable
+    private OpenSprinklerHttpInterfaceConfig openSprinklerConfig;
+
+    public OpenSprinklerHttpBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        OpenSprinklerHttpInterfaceConfig openSprinklerConfig = getConfig().as(OpenSprinklerHttpInterfaceConfig.class);
+        this.openSprinklerConfig = openSprinklerConfig;
+
+        logger.debug("Initializing OpenSprinkler with config (Hostname: {}, Port: {}, Password: {}, Refresh: {}).",
+                openSprinklerConfig.hostname, openSprinklerConfig.port, openSprinklerConfig.password,
```

Comment:
Didn't notice before: please do not log passwords.

Reply:
Holy s***! Thanks for catching, that was code I simply took over from the current version of the binding, I should've taken a deeper look here :(